editor/editor_properties.cpp

```diff
@@ -2182,8 +2182,97 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		} break;
 		case OBJ_MENU_NEW_SCRIPT: {
 
-			if (Object::cast_to<Node>(get_edited_object())) {
+			if (Object::cast_to<Node>(get_edited_object()) &&
+					EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->is_a_parent_of(Object::cast_to<Node>(get_edited_object())) &&
+					get_edited_property() == "script") {
+				// This is a script property of a node in the scene tree, so we should defer to the SceneTreeDock's script creation code
 				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+			} else {
+
+				// Based upon editor/scene_tree_dock.cpp#_tool_selected (case TOOL_ATTACH_SCRIPT); edited
+				// to account for exported script properties, resources, and sub-resources
```

Comment:
I would want to try to merge this again, but instead of merging it back into the mentioned function, I would prefer to merge it into `editor/editor_node.cpp` in some manner.

That said, the reason I have not done so already is that I don't think that's within the scope of this PR -- but if it can be considered within the scope, I might try my hand at it.